tests/unittests/Makefile

```diff
@@ -0,0 +1,60 @@
+#
+# Copyright (c) Microsoft. All rights reserved
+#
+
+include $(CLEAR_VARS)
+
+CLANG := clang++
+
+SRCDIR := Foundation
+SRCFILES := $(wildcard $(SRCDIR)/*.m*)
+RF_SRCFILES := $(wildcard $(SRCDIR)/ReferenceFoundation/*.mm)
+
+OUTDIR := ../../build/Tests/UnitTests/Foundation/OSX
+OUTTARGETS := $(addsuffix .o, $(addprefix $(OUTDIR)/, $(basename $(notdir $(SRCFILES)))))
+OUTTARGETS += $(addsuffix .o, $(addprefix $(OUTDIR)/ReferenceFoundation_, $(basename $(notdir $(RF_SRCFILES)))))
+
+SRC := Foundation/NSArrayTests.mm
+
+INC := ./Foundation/ReferenceFoundation ../frameworks/include ../frameworks/gtest ../frameworks/gtest/include 
+INC += ../frameworks/OSXShims/include Foundation/ReferenceFoundation ../../frameworks/include
+INC_PARAMS := $(INC:%=-I %)
+
+CFLAGS += -std=c++14
+CFLAGS += -x objective-c++
+CFLAGS += $(INC_PARAMS)
+
+Shims : $(OUTDIR)/windows.o $(OUTDIR)/IwMalloc.o
```

Comment:
>Shims [](start = 0, length = 5)

Shims should possibly be phony, otherwise a file named Shims will confuse/enrage it.